src/osha/oira/ploneintranet/quaive_view.py

```diff
@@ -5,17 +5,43 @@
 from euphorie.content.utils import ToolTypesInfo
 from logging import getLogger
 from plone import api
+from plone.base.utils import base_hasattr
+from plone.dexterity.interfaces import IDexterityFTI
 from plone.memoize.view import memoize
 from plone.memoize.view import memoize_contextless
 from plone.supermodel.model import SchemaClass
 from Products.Five import BrowserView
 from zope.component import getUtility
+from zope.component import queryUtility
 from zope.interface import providedBy
 
 
 logger = getLogger(__name__)
 
 
+class QuaiveHelpers(BrowserView):
+
+    @property
+    @memoize
+    def allowed_content_types(self):
+        """Which content types are addable on the context?
+
+        Main use case: check if at least one content type can be added.
+        Otherwise there is no need to show an Add menu.
+        """
+        fti = queryUtility(IDexterityFTI, name=self.context.portal_type)
+        if fti is None:
+            return []
+        return fti.allowed_content_types
```

Comment:
This is generally too permissive as it just checks the FTI definition without taking into account constraints and permissions.

Reply:
I have fixed this in commit 76a63ff2962dd9f2d3d1d3e84fad72012f2177e8.